### PR TITLE
Fix dockerfile hadolint problems

### PR DESCRIPTION
**Pull request by potiuk**
The hadolint check only checked the "main dir" Dockerfile
but we have more of them now. All of them are now checked.

The following problems are fixed:

 * DL3000 Use absolute WORKDIR
 * DL4000 MAINTAINER is deprecated
 * DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it.
 * SC2046 Quote this to prevent word splitting.

The followiing problems are ignored:

 * DL3018 Pin versions in apk add. Instead of `apk add <package>` use `apk add
   <package>=<version>`

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

File: chart/dockerfiles/pgbouncer/Dockerfile [hunk 1]
```diff
@@ -15,19 +15,20 @@
 # specific language governing permissions and limitations
 # under the License.
 ARG ALPINE_VERSION="3.12"
-
 FROM alpine:${ALPINE_VERSION} AS builder
+SHELL ["/bin/ash", "-eo", "pipefail", "-c"]
```
Comment (kaxil) [inline chart/dockerfiles/pgbouncer/Dockerfile]:
```suggestion
SHELL ["/bin/bash", "-eo", "pipefail", "-c"]
```

Comment (potiuk) [reply chart/dockerfiles/pgbouncer/Dockerfile]: Interesting mistake :)

**Comment by potiuk** [comment]
I need the #10750 to be meerged first :(

File: chart/dockerfiles/pgbouncer/Dockerfile (hunk 1 above)
Comment (kaxil) [reply chart/dockerfiles/pgbouncer/Dockerfile]: This hasn't been resolved, still says bash

**Comment by kaxil** [comment]
Can you please add a OR description, I would love to help the community understand what the PR actually fixes. As little as "fixes formatting" (just an example) is helpful for the community to understand when they see the PR

**Comment by potiuk** [comment]
> Can you please add a PR description, I would love to help the community understand what the PR actually fixes. As little as "fixes formatting" (just an example) is helpful for the community to understand when they see the PR

Good Idea. Added. I also made comments in the Dockerfiles where iItook deliberate version of ignoring a rule explaining why I did so.

**Comment by potiuk** [comment]
Also added  explanation what each ignored rule means and link to the wiki describing them .

**Pull request merged**